infinispan-remote/src/main/java/org/hibernate/ogm/datastore/infinispanremote/impl/InfinispanRemoteStoredProceduresManager.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.datastore.infinispanremote.impl;
+
+import java.lang.invoke.MethodHandles;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.hibernate.ogm.datastore.infinispanremote.logging.impl.Log;
+import org.hibernate.ogm.datastore.infinispanremote.logging.impl.LoggerFactory;
+import org.hibernate.ogm.dialect.query.spi.ClosableIterator;
+import org.hibernate.ogm.model.spi.Tuple;
+import org.hibernate.ogm.storedprocedure.ProcedureQueryParameters;
+import org.hibernate.ogm.util.impl.CollectionHelper;
+import org.hibernate.ogm.util.impl.TupleExtractor;
+
+import org.infinispan.client.hotrod.RemoteCache;
+import org.infinispan.client.hotrod.exceptions.HotRodClientException;
+
+/**
+ * @author The Viet Nguyen &amp;ntviet18@gmail.com&amp;
+ */
+public class InfinispanRemoteStoredProceduresManager {
+
+	private static final Log log = LoggerFactory.make( MethodHandles.lookup() );
+
+	private static final Pattern REFERENCE_ERROR_REGEXP = Pattern.compile( ".*\"([a-zA-Z_$]+)\" is not defined in <eval>.*" );
```

Comment:
It is fine relying on regex on **ISPN 027002**, but I think we can not rely on message text. In general it could change or be different in other languages.

Reply:
That's right, logger IDs should be maintained stable across versions (and not reused). But the message text might be updated and could be translated.

Reply:
A general workaround if you can't parse it is to report the original datastore error message into upfront Hibernate OGM message.

Reply:
@fax4ever , @Sanne , @schernolyas , I think it's the only way we know whether we used an undefined parameters or not. Because Infinispan at the moment doesn't seem to parse Nashorn errors properly.

> Caused by: org.infinispan.client.hotrod.exceptions.HotRodClientException:Request for messageId=5718 returned server error (status=0x85): javax.script.ScriptException: ReferenceError: "param" is not defined in <eval> at line number 2
> <eval>:2 ReferenceError: "param" is not defined
> 	at org.infinispan.client.hotrod.impl.protocol.Codec20.checkForErrorsInResponseStatus(Codec20.java:332)
> 	at org.infinispan.client.hotrod.impl.protocol.Codec20.readHeader(Codec20.java:178)
> 	at org.infinispan.client.hotrod.impl.transport.netty.HeaderDecoder.decode(HeaderDecoder.java:138)
> 	at 

According to `int SERVER_ERROR_STATUS = 0x85;`. The status tell us nothing about the actual cause.

Reply:
Unfortunately Hot Rod return the same code in both situations. Anyway a possible wrong interpretation of the error would not cause unexpected errors, because the parsing in inside the catch. Temporary I think that you can put TODO, just for the moment. WDYT @DavideD?

Reply:
It's annoying that we cannot get the exact error but I don't think this issue is blocking. Let's just make sure that the tests we have check that an exception is thrown and then we can work it out later how to make it user friendly

Reply:
@ntviet18 thanks for the explanation, I didn't know that. Ok then let's keep this for now - we can ask the Infinispan team to improve on this and then we will be able to improve OGM.

Reply:
(I agree with you all)